hw/ip/otbn/doc/_index.md

```diff
@@ -372,14 +372,38 @@ The number of entries in the loop stack is implementation-dependent.
 
 ## Design Details
 
-<div class="bd-callout bd-callout-warning">
-  <h5>Note</h5>
+### Error Handling and Reporting
 
-  To be filled in as we create the implementation.
-</div>
+By design, OTBN is a simple processor and provides no error handling support to code that runs on it.
+Any errors will cause OTBN to stop executing and report the failure to the host processor.
+
+Whenever OTBN observes an error, it will generate an alert.
+This gets sent to the alert manager.
+The alert will either be fatal or recoverable, depending on the class of error: see [Alerts]({{< relref "hw/ip/otbn/doc#alerts" >}}) and [Error Conditions]({{< relref "hw/ip/otbn/doc#error-conditions" >}}) below for details.
+
+If OTBN was running when the alert occurred (this is true whenever {{< regref "STATUS.busy" >}} is high), it will also:
+- Immediately stop fetching and executing instructions.
+- Set {{< regref "INTR_STATE.done" >}} and clear {{< regref "STATUS.busy" >}}, marking the operation as completed.
+- Set the {{< regref "ERR_CODE" >}} register to a non-zero value describing the error.
+
+Note that OTBN can detect some errors even when it isn't running.
+One example of this is an error caused by an ECC failure when reading or programming OTBN's memories over the bus.
+In this case, the {{< regref "ERR_CODE" >}} register will not change.
+This avoids race conditions with the host processor's error handling software.
 
-By design, OTBN is a simple processor and has essentially no error handling support.
-When anything goes wrong (an out-of-bounds memory operation, an invalid instruction encoding, etc.), OTBN will stop fetching instructions, and set the `ERR_CODE` register and the `err` bit of the `INTR_STATE` register.
+### Alerts
+
+OTBN has two alerts, one recoverable and one fatal.
+The [Error Conditions]({{< relref "hw/ip/otbn/doc#error-conditions" >}}) section has a detailed list of error conditions and the alerts that they raise.
+
+A **recoverable alert** is a one-time triggered alert for recoverable error conditions.
+These is only triggered when OTBN is running, so will always imply a write to {{< regref "ERR_CODE." >}}
```

Comment:
```suggestion
It is only triggered when OTBN is running, so will always imply a write to {{< regref "ERR_CODE." >}}
```

Reply:
Well, "these is" is definitely wrong, good spot! I think I probably prefer the plural though: I think I'd say "*A farmyard animal is an animal that lives on a farm. These animals are very cute.*", rather than "*A farmyard animal is an animal that lives on a farm. This animal is very cute.*".

But let's side-step the issue and replace "These/It is" with "Recoverable alerts are". It's probably clearer anyway.